src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.CoreRT.cs

```diff
@@ -331,9 +362,67 @@ public object GetOrRegisterObjectForComInstance(IntPtr externalComObject, Create
             if (flags.HasFlag(CreateObjectFlags.Aggregation))
                 throw new NotImplementedException();
 
-            object? wrapperMaybeLocal = wrapperMaybe;
-            retValue = null;
-            throw new NotImplementedException();
+            if (flags.HasFlag(CreateObjectFlags.Unwrap))
+            {
+                var comInterfaceDispatch = TryGetComInterfaceDispatch(externalComObject);
+                if (comInterfaceDispatch != null)
+                {
+                    retValue = ComInterfaceDispatch.GetInstance<object>(comInterfaceDispatch);
+                    return true;
+                }
+            }
+
+            if (!flags.HasFlag(CreateObjectFlags.UniqueInstance))
+            {
+                using (LockHolder.Hold(_lock))
+                {
+                    if (_rcwCache.TryGetValue(externalComObject, out var handle))
```

Comment:
```suggestion
                    if (_rcwCache.TryGetValue(externalComObject, out GCHandle handle))
```
Nit - coding conventions